src/mapblock_mesh.cpp

```diff
@@ -321,39 +316,56 @@ u16 getSmoothLight(v3s16 p, v3s16 corner, MeshMakeData *data)
 	return getSmoothLightCombined(p, data);
 }
 
-/*
-	Converts from day + night color values (0..255)
-	and a given daynight_ratio to the final SColor shown on screen.
-*/
-void finalColorBlend(video::SColor& result,
-		u8 day, u8 night, u32 daynight_ratio)
-{
-	s32 rg = (day * daynight_ratio + night * (1000-daynight_ratio)) / 1000;
+void getSunlightColor(video::SColor &sunlight, u32 daynight_ratio){
+	s32 rg = (255 * daynight_ratio) / 1000;
 	s32 b = rg;
 
 	// Moonlight is blue
-	b += (day - night) / 13;
-	rg -= (day - night) / 23;
+	b += (255) / 13;
+	rg -= (255) / 23;
 
 	// Emphase blue a bit in darker places
 	// Each entry of this array represents a range of 8 blue levels
 	static const u8 emphase_blue_when_dark[32] = {
 		1, 4, 6, 6, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0, 0,
 		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 	};
+
 	b += emphase_blue_when_dark[irr::core::clamp(b, 0, 255) / 8];
 	b = irr::core::clamp(b, 0, 255);
-
-	// Artificial light is yellow-ish
-	static const u8 emphase_yellow_when_artificial[16] = {
-		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 10, 15, 15, 15
-	};
```

Comment:
I have not checked if the displayed color is still exactly the same, probably not. But artificial light is still yellow-ish. See the constant artificial light color in the shaders and in `finalColorBlend()`.